sql/item.cc

```diff
@@ -10844,6 +10845,106 @@ const char *dbug_print_item(Item *item)
     return "Couldn't fit into buffer";
 }
 
+#define DBUG_BIG_BUF_SIZE 20480
+static char big_buf[DBUG_BIG_BUF_SIZE];
+
+const char *dbug_print_items(Item *item);
+void dbug_add_print_items(Item *item);
+void dbug_add_print_items(List<Item> &list)
+{
+  List_iterator<Item> li(list);
+  Item *sub_item;
+  strcpy( big_buf + strlen( big_buf ), "Children:" );
+  while ((sub_item = li++))
+	if( strlen( big_buf ) < DBUG_BIG_BUF_SIZE - DBUG_PRINT_SIZE - 7 )
+	{
+	  sprintf( big_buf + strlen( big_buf ), "[%p: ", (void*)sub_item );
+	  dbug_add_print_items(sub_item);
+	  sprintf( big_buf + strlen( big_buf ), "]" );
+	}
+	else
+	{
+	  strcat( big_buf, "..." );
+	  break;
+	}
+  return;
+}
+
+const char *dbug_print_items(List<Item> &list)
+{
+  asm ("");		// stop optimizer from removing this function
+  *big_buf = 0;
+
+  dbug_add_print_items( list );
+  return big_buf;
+}
+
+void dbug_add_print_items(Item *item)
+{
+  asm ("");		// stop optimizer from removing this function
+  if( item )
+  {
+	strncpy( big_buf + strlen( big_buf ), dbug_print_item(item), DBUG_BIG_BUF_SIZE );
+	List_iterator<Item> li;
+	bool show_children = false;
+
+	switch( item->type() )
+	{
+	case Item::COND_ITEM:
+	  li = *((Item_cond*) item)->argument_list();
+	  show_children = true;
+	  break;
+	default:
+	  ;
+	}
+
+	if( show_children )
+	{
+	  Item *sub_item;
+	  strcat( big_buf + strlen( big_buf ), ".Children:" );
```

Comment:
looks like a pointer should be maintained in the function. See also `strnmov` (in codebase).